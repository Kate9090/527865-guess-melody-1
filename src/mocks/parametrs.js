const params = {
  gameTime: 5,
  maxMistakes: 3,
};

export default params;
