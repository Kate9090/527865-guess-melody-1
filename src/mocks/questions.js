const questions = [
  {
    type: `genre`,
    genre: `rock`,
    gameTime: 5,
    errorCount: 3,
    answers: [
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `rock`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `pop`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `jazz`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `blues`,
      },
    ],
  },
  {
    type: `artist`,
    artist: `Пелагея`,
    gameTime: 5,
    errorCount: 3,
    answers: [
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.png`,
        artist: `Пелагея`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.png`,
        artist: `Краснознаменная дивизия имени моей бабушки`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.png`,
        artist: `Lordi`,
      },

    ],
  }
];

export default questions;
