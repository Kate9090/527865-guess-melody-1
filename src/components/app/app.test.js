import React from 'react';
import renderer from 'react-test-renderer';

import App from './app.jsx';

import questions from '../../mocks/questions';
import parametrs from '../../mocks/parametrs';

const mockQuestion = questions;
const mockParams = parametrs;

it(`renders correctly app screen`, () => {
  const tree = renderer
    .create(<App
      gameParams={mockParams}
      questions={mockQuestion}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
