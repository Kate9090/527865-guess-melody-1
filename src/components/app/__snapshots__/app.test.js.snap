// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly app screen 1`] = `
Array [
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    />
  </header>,
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      rock
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={0}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="rock"
          />
          <label
            className="game__check"
            htmlFor={0}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={1}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="pop"
          />
          <label
            className="game__check"
            htmlFor={1}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={2}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="jazz"
          />
          <label
            className="game__check"
            htmlFor={2}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={3}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="blues"
          />
          <label
            className="game__check"
            htmlFor={3}
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>,
]
`;
